sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include "net/eui64.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/ndp.h"
+#include "net/gnrc/ndp/internal.h"
+#include "net/gnrc/sixlowpan.h"
+#include "net/gnrc/sixlowpan/ctx.h"
+#include "random.h"
+#include "timex.h"
+
+#include "net/gnrc/sixlowpan/nd.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static inline void _rtr_sol_reschedule(gnrc_ipv6_netif_t *iface, uint32_t sec_delay)
+{
+    vtimer_remove(&iface->rtr_sol_timer);
+    vtimer_set_msg(&iface->rtr_sol_timer, timex_set(sec_delay, 0), gnrc_ipv6_pid,
+                   GNRC_SIXLOWPAN_ND_MSG_MC_RTR_SOL, iface);
+}
+
+static inline uint32_t _binary_exp_backoff(uint32_t base_sec, unsigned int exp)
+{
+    return genrand_uint32_range(0, (1 << exp) - 1) * base_sec;
+}
+
+static inline void _revert_iid(uint8_t *iid)
+{
+    iid[0] ^= 0x02;
+}
```

Comment:
Personally, I don't see a need for these inline functions, but at the end it's a matter of taste.